### ISSUE TITLE
Rubberband effect and status bar/nav bar colors

### ISSUE THREAD

**Issue by Jejebond** [Nerwyn/material-rounded-theme#14]
**Is your feature request related to a problem? Please describe.**
When it come to get as close as possible to Google Home app, i can see 2 details :
1)Rubber band effect when scrolling on limitations
2)Top status bar and bottom nav bar colors

**Describe the solution you'd like**
1)Rubber band effect should be avoided on homme assistant app nav bar
2)Top status bar should be same as background
3)Bottom nav bar should be same color app nav bar

**Describe alternatives you've considered**
Android 14 comes with a solution that is not working for Bottom nav bar color in dev settings, it's called transparent navigation bar.

**Additional context**
Probably related to companion app and not themes, but is there something that can be done ?

VIDEOS

https://github.com/Nerwyn/material-rounded-theme/assets/3942165/128d22c0-ee96-4445-a179-e3d96f1ab425

https://github.com/Nerwyn/material-rounded-theme/assets/3942165/a0602f75-5ff9-46a9-ace3-dc054e47bebf

https://github.com/Nerwyn/material-rounded-theme/assets/3942165/9f3943e2-d360-4293-ba15-635d64cf896d

**Comment by Jejebond** [comment]
Colors for nav bar and statuts bar can be fixed :
[Theming - Companion](https://companion.home-assistant.io/docs/integrations/theming/)

```
Colors should be specified in hex format (e.g. #0099ff) and defining element colors through variable names is not supported.

primary-background-color for the navigation bar background color Android
app-header-background-color for the status bar background color Android
iOS

As of version 2020.3, the iOS app will accept colors specified in hex, rgb, hsl, rgba, hsla formats or using a valid [HTML color name](https://www.w3schools.com/colors/colors_names.asp); although formats with alpha values are recognized, using alpha values less than 100 % (or 1) will currently lead to a color mismatch. 2020.2 and earlier versions of the app require colors to be specified in hex.

primary-background-color for the background color of the web view iOS
app-header-background-color for the status bar background color iOS
primary-color for the pull-to-refresh control/spinner
```

**Comment by Nerwyn** [comment]
We're a bit limited on what we can and cannot do with Home Assistant themes, and things like the toolbar are the result of card-mod and CSS hacks to add pseudo elements and backgrounds to existing elements. I don't think I can fix the rubber banding effect (but will dig more into it) but that bit on navigation and status bar theming is a good find! Let me see if I can make use of it. The incorrect navigation and status bar colors had also been bugging me for a while but I assumed it wasn't fixable.

**Comment by Jejebond** [comment]
Ok, i've done some testing :

**primary-background-color :** Sadly, it acts for **background app** AND **Navigation bar**... that's a companion app issue. But as we want a bottom app header bar, it breaks that continuity.
**app-header-background-color :** Again, saldy, it acts for **app header** AND **Status bar**.

Companion app is built in therms of coloring for a top app-header 😢.

The only way is to ask for an enhancement to companion app... if it does not already exist. Except if an "bottom bar" addon exist to fix that.

Same for rubber effect. The rubber effect is also present on native home assistant theme.

**Comment by Nerwyn** [comment]
You beat me to it. Except that the dashboard background uses `lovelace-background`, which normally points to `primary-background-color` but we can instead make the app color and reassign `primary-background-color` to be the same as the toolbar color. This does also make all of the settings pages the same color as the toolbar but I think it's a good tradeoff.

`app-header-background-color` is hard to get around as you said. But I think I may be able to do something with card-mod.

**Comment by Nerwyn** [comment]
I'm also going to have to adjust how I assign these colors because I imagine fixing this in the material you toolbar version will break the colors in the no mod version of the theme.

**Comment by Jejebond** [comment]
Nice. YOU beat me to it xD.

I'm using alpha if you want to deploy some test versions. i'll give you a feedback.

**Comment by Nerwyn** [comment]
I see that you are using the alpha v2 material you color options! I've got to copy a lot of the updated code to v1 since this fix should go in that as well so it'll take me a bit to have a usable alpha/beta version.

**Comment by Nerwyn** [comment]
@Jejebond I've put out a new v2 alpha which should fix the navbar and toolbar colors. Give it a try! I'm going to port over a bunch of updates to v1 and create a beta patch version for that as well.

**Comment by Nerwyn** [comment]
One problem with these changes is that the navbar color is now the same as the status bar in the no mod versions of the theme. Considering that this theme was made to be used with card mod, I'm willing to let that slide in order to improve the card-mod bottom toolbar main version.

**Comment by Nerwyn** [comment]
As for overscroll stretch - From what I can find setting `overscroll-stretch` to none in the header styles is supposed to fix it but it looks like Home Assistant's companion app reads the theme CSS before card-mod is applied, so I can't fix it. It's also an issue without my theme, so you may want to create a bug for the Android app instead.

**Comment by Jejebond** [comment]
> One problem with these changes is that the navbar color is now the same as the status bar in the no mod versions of the theme. Considering that this theme was made to be used with card mod, I'm willing to let that slide in order to improve the card-mod bottom toolbar main version.

True, but its still better. I agree with you.

> As for overscroll stretch - From what I can find setting `overscroll-stretch` to none in the header styles is supposed to fix it but it looks like Home Assistant's companion app reads the theme CSS before card-mod is applied, so I can't fix it. It's also an issue without my theme, so you may want to create a bug for the Android app instead.

Sure ;)

The theme looks stunning !

![Screenshot_20240503-212745](https://github.com/Nerwyn/material-rounded-theme/assets/3942165/70c0f790-71ef-48a1-91d8-ccf2a9d6416d)

I'll focus on reporting Material you issue from now on.

> **TODO** before release- Update README screenshots, and use more material symbols.

PS: about Material symbols, i've stoopped using those as sometimes they won't load as fast as mdi library ones. Witch result in blank icon. Should i open a new issue ? i thought it wasn't linked to your theme.

**Comment by Nerwyn** [comment]
I've fixed the no-mod theme colors so the v1 version is ready for release once I update the README.

Material symbols isn't linked to my theme, but as we're both working on bringing material you styling to Home Assistant I want to give them free advertising. The creator of it was also one of the first people to show support for this theme. Have you tried the latest v1 release? It's README also has [instructions for adding it has a resource](https://github.com/beecho01/material-symbols?tab=readme-ov-file#add-hacs-custom-repository) which should improve it's performance. It loads just as fast as MDI icons for me.

**Comment by Nerwyn** [comment]
Status and navbar colors fixed in [1.6.0](https://github.com/Nerwyn/material-rounded-theme/releases/tag/1.6.0), and added to 2.0.0 as of alpha.007
